internal/kafka/internal/handlers/cluster.go

```diff
@@ -0,0 +1,88 @@
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/acl"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/client/observatorium"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/handlers"
+)
+
+type clusterHandler struct {
+	kasFleetshardOperatorAddon                           services.KasFleetshardOperatorAddon
+	observabilityConfiguration                           *observatorium.ObservabilityConfiguration
+	clusterService                                       services.ClusterService
+	enterpriseClusterRegistrationAccessControlListConfig *acl.EnterpriseClusterRegistrationAccessControlListConfig
+}
+
+func NewClusterHandler(kasFleetshardOperatorAddon services.KasFleetshardOperatorAddon, observabilityConfiguration *observatorium.ObservabilityConfiguration,
+	clusterService services.ClusterService, enterpriseClusterRegistrationAccessControlListConfig *acl.EnterpriseClusterRegistrationAccessControlListConfig) *clusterHandler {
+	return &clusterHandler{
+		kasFleetshardOperatorAddon:                           kasFleetshardOperatorAddon,
+		observabilityConfiguration:                           observabilityConfiguration,
+		clusterService:                                       clusterService,
+		enterpriseClusterRegistrationAccessControlListConfig: enterpriseClusterRegistrationAccessControlListConfig,
+	}
+}
+
+func (h clusterHandler) RegisterEnterpriseCluster(w http.ResponseWriter, r *http.Request) {
+	var clusterPayload public.EnterpriseOsdClusterPayload
+
+	ctx := r.Context()
+
+	cfg := &handlers.HandlerConfig{
+		MarshalInto: &clusterPayload,
+		Validate: []handlers.Validate{
+			handlers.ValidateLength(&clusterPayload.ClusterId, "cluster id", ClusterIdLength, &ClusterIdLength),
+
+			handlers.ValidateExternalClusterId(&clusterPayload.ClusterExternalId, "external cluster id"),
+
+			handlers.ValidateClusterId(&clusterPayload.ClusterId, "cluster id"),
+
+			ValidateClusterIdIsUnique(&clusterPayload.ClusterId, h.clusterService),
+
+			handlers.ValidateDnsName(&clusterPayload.ClusterDnsName, "cluster dns name"),
+		},
+		Action: func() (interface{}, *errors.ServiceError) {
+
+			claims, claimsErr := getClaims(ctx)
+
+			if claimsErr != nil {
+				return nil, claimsErr
+			}
+
+			orgId, getOrgIdErr := claims.GetOrgId()
+
+			if getOrgIdErr != nil {
+				return nil, errors.GeneralError(getOrgIdErr.Error())
+			}
+
+			isOrgAccepted := h.enterpriseClusterRegistrationAccessControlListConfig.EnterpriseClusterRegistrationAccessControlList.IsOrganizationAccepted(orgId)
+
+			if !isOrgAccepted {
+				return nil, errors.GeneralError("organization '%s' is not authorized to register enterprise cluster", orgId)
+			}
+
+			clusterRequest := &api.Cluster{
+				ClusterType: api.Enterprise.String(),
+				Status:      api.ClusterAccepted,
```

Comment:
Is accepted the right status? If we're not reconciling the cluster, it won't ever reach another status.

Reply:
nvw. I see that you are updating the status in the reconcilers